assets/js/blocks/cart-checkout/subscriptions-integration.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import {
+	ExperimentalOrderShippingPackages,
+	ShippingRatesControlPackage,
+} from '@woocommerce/blocks-checkout';
+import { registerPlugin } from '@wordpress/plugins';
+import { useStoreCart, useSelectShippingRate } from '@woocommerce/base-hooks';
+import { useMemo } from '@wordpress/element';
+
+const RenderSubscriptionPackages = () => {
+	const {
+		extensions: { subscriptions },
+	} = useStoreCart();
+
+	// Flatten all packages from recurring carts.
+	const packages = useMemo( () => {
+		const newPackages = [];
+
+		Object.values( subscriptions ).forEach( ( recurringCart ) => {
+			const recurringCartPackages = recurringCart.shipping_rates || [];
+
+			recurringCartPackages.forEach( ( recurringCartPackage ) => {
+				newPackages.push( recurringCartPackage );
+			} );
+		} );
+
+		return newPackages;
+	}, [ subscriptions ] );
+
+	const { selectShippingRate, selectedShippingRates } = useSelectShippingRate(
+		packages
+	);
```

Comment:
I'm not sure I understand `useSelectShippingRate`, does it allow selecting a rate from a single package, or a rate from different packages?

Also, is there any value of having it this high in the tree? can it be moved inside  `ShippingRatesControlPackage` or be supplied from `ExperimentalOrderShippingPackages` as a `fillProps`?

Reply:
It manages selected state across multiple packages, as it did prior to this PR. In this context we are using it to manage state of the multiple packages subscriptions can return, separate from the multiple packages core can return for regular items.

Reply:
Would it make sense to limit it to a single package?

Reply:
The selected rates I think are exposed in other places... If we do it per package that will make that more difficult?

Side note, I actually wanted to remove selected methods state from this hook competely but had some pushback earlier in the week. Happy to refactor further - just say so.

Reply:
> The selected rates I think are exposed in other places... If we do it per package that will make that more difficult?

I assume selected rates are needed for express payments? can't they be derived from cart response?

Reply:
@senadir This is what it took to move it to package level and remove from the subscriptions integration. We track state at package level, but the select shipping rates hook now has no state so that it can simply make requests and determine if a request is being made. Select rates is derived where needed.

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3687/commits/5b7f860113fb7ad3d6741edfabf00e562d57253a